shipment_advice_planner/models/shipment_advice_planner.py

```diff
@@ -0,0 +1,110 @@
+        return defaults
+
+    def button_plan_shipments(self):
+        self.ensure_one()
+        shipment_advices = self._plan_shipments_for_method(
+            self.picking_to_plan_ids, self.shipment_planning_method
+        )
+        if not shipment_advices:
+            return {}
+        return {
+            "type": "ir.actions.act_window",
+            "name": _("Shipment Advice"),
+            "view_mode": "tree,form",
+            "res_model": shipment_advices._name,
+            "domain": [("id", "in", shipment_advices.ids)],
+            "context": self.env.context,
+        }
+
+    @api.model
+    def _plan_shipments_for_method(
+        self, pickings_to_plan, shipment_planning_method, extra_values=False
+    ):
+        prepare_method_name = self._get_prepare_method_name(shipment_planning_method)
+        if not hasattr(self, prepare_method_name):
+            raise NotImplementedError(
+                _("There is no implementation for the planning method '%s'")
+                % shipment_planning_method
+            )
+        prepare_method = getattr(self, prepare_method_name)
+        shipment_advice_model = self.env["shipment.advice"]
+        create_vals = []
+        for (
+            warehouse,
+            pickings_to_plan,
+        ) in self._get_picking_to_plan_by_warehouse(pickings_to_plan).items():
+            create_vals.append(
+                prepare_method(warehouse, pickings_to_plan, extra_values)
+            )
+        return shipment_advice_model.create(create_vals)
+
+    @api.model
+    def _get_prepare_method_name(self, shipment_planning_method):
+        return f"_prepare_shipment_advice_{shipment_planning_method}_vals"
+
+    @api.model
+    def _get_picking_to_plan_by_warehouse(self, pickings_to_plan):
+        warehouse_model = self.env["stock.warehouse"]
+        picking_model = self.env["stock.picking"]
+        res = {}
+        for group in picking_model.read_group(
+            [("id", "in", pickings_to_plan.ids)],
+            ["warehouse_id"],
+            ["warehouse_id"],
+        ):
+            warehouse = warehouse_model.browse(group.get("warehouse_id")[0])
+            res[warehouse] = picking_model.search(group.get("__domain"))
+        return res
+
+    @api.model
+    def _prepare_shipment_advice_simple_vals(
+        self, warehouse, pickings_to_plan, extra_values=False
+    ):
+
+        vals = self._prepare_shipment_advice_common_vals(warehouse, extra_values)
+        vals["planned_move_ids"] = [Command.set(pickings_to_plan.move_ids.ids)]
+        return vals
+
+    @api.model
+    def _prepare_shipment_advice_common_vals(self, warehouse, extra_values=False):
+
+        values = {
+            "shipment_type": "outgoing",
+            "warehouse_id": warehouse.id,
+            "company_id": warehouse.company_id.id,
+        }
+        if isinstance(extra_values, dict):
+            values.update(extra_values)
+        return values
```

Comment:
@sbejaoui The prepare method should not be `api.model` method but ensure one record. We could imagine that some extension could add specific fields into the wizard and complete the prepared values with these fields.